### PR TITLE
add parsable decorator

### PR DESCRIPTION
**Pull request by olliethomas**

### CODE REVIEW

File: kit/typing.py [hunk 1]
```diff
@@ -22,3 +22,15 @@ def __call__(self, func: _F) -> _F:
         super_method = getattr(self.interface, func.__name__, None)
         assert super_method is not None, f"'{func.__name__}' does not exist in {self.interface}"
         return func
+
+
+class parsable:  # pylint: disable=invalid-name
+    """Mark an object's __init__ as parsable by Configen, so only 3.7 type annotations should be used."""
```
Comment (thomkeh) [inline kit/typing.py:28]: Something like this seems more accurate:

```suggestion
    """Mark an object's __init__ as parsable by Configen, so only pre-3.9 type annotations should be used."""
```

File: kit/typing.py [hunk 2]
```diff
@@ -22,3 +22,15 @@ def __call__(self, func: _F) -> _F:
         super_method = getattr(self.interface, func.__name__, None)
         assert super_method is not None, f"'{func.__name__}' does not exist in {self.interface}"
         return func
+
+
+class parsable:  # pylint: disable=invalid-name
+    """Mark an object's __init__ as parsable by Configen, so only pre-3.9 type annotations should be used."""
+
+    def __init__(self):
```
Comment (thomkeh) [inline kit/typing.py:30]: Could just leave out this init

**Comment by olliethomas** [comment]
@thomkeh how to inspect an argument's type to check that the correct syntax is used?

File: kit/typing.py [hunk 3]
```diff
@@ -3,6 +3,8 @@
 
 __all__ = ["implements"]
 
+from typing_extensions import get_type_hints
```
Comment (thomkeh) [inline kit/typing.py:6]:
```suggestion
from typing import get_type_hints
```

File: kit/typing.py [hunk 4]
```diff
@@ -22,3 +24,15 @@ def __call__(self, func: _F) -> _F:
         super_method = getattr(self.interface, func.__name__, None)
         assert super_method is not None, f"'{func.__name__}' does not exist in {self.interface}"
         return func
+
+
+def parsable(func: _F) -> _F:
+    """Mark an object's __init__ as parsable by Configen, so only pre-3.9 type annotations should be used."""
+    assert func.__name__ == "__init__", "@parsable can only be used to decorate __init__."
```
Comment (thomkeh) [inline kit/typing.py:31]: does configen not work with functions?

File: tests/typing_test.py [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+from kit.typing import parsable
+
+
+def test_parsable() -> None:
+    class Foo:
+        @parsable
+        def __init__(self) -> None:
+            ...
+
+    _ = Foo()
```
Comment (thomkeh) [inline tests/typing_test.py:15]: not really needed

File: tests/typing_test.py [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+
+
+def test_parsable_valid() -> None:
+    class Foo:
+        @parsable
+        def __init__(self, a: int, b: Union[int, float], c: List[str]):
+            ...
+
+    _ = Foo(a=1, b=1.2, c=["bar"])
```
Comment (thomkeh) [inline tests/typing_test.py:24]: same

File: kit/typing.py (hunk 4 above)
Comment (thomkeh) [reply kit/typing.py:31]: seems not

File: tests/typing_test.py (hunk 5 above)
Comment (thomkeh) [reply tests/typing_test.py:15]: (the instantiation of the class, that is)

File: kit/typing.py (hunk 4 above)
Comment (olliethomas) [reply kit/typing.py:31]: I don't know. I think we only use it with objects

File: kit/typing.py (hunk 3 above)
Comment (olliethomas) [reply kit/typing.py:6]: isn't PalKit for 3.7+?

Comment (thomkeh) [reply kit/typing.py:6]: yes? `get_type_hints()` has been in the standard library since python 3.5

Comment (thomkeh) [reply kit/typing.py:6]: maybe you're confusing it with `get_args()` or `get_origin()` which were added in 3.8, I think.

**Pull request merged**